uservice_ccutter/server.py

```diff
@@ -104,88 +104,103 @@ def post_request(ptype):
+            # replace cookiecutter.json
+            with open(clonedir + "/cookiecutter.json", "w") as ccf:
+                ccf.write(json.dumps(userdict, indent=4))
+            try:
+                cookiecutter(clonedir, no_input=True)
+            except (CookiecutterException, TypeError) as exc:
+                raise BackendError(status_code=500,
+                                   reason="Internal Server Error",
+                                   content="Project creation failed: " +
+                                   str(exc))
+            # Here we make the assumption that cookiecutter created
+            #  a single directory.  For our projects, that is a good
+            #  assumption, usually: a git repo has a unique top-level
+            #  directory, and a-thing-to-be-committed-by-git is what
+            #  we are using cookiecutter for.
+            flist = os.listdir(tgtdir)
+            if not flist:
+                raise BackendError(status_code=500,
+                                   reason="Internal Server Error",
+                                   content="No project created")
+            gitbase = flist[0]
+            gitdir = tgtdir + "/" + gitbase
+            os.chdir(gitdir)
+            userdict["local_git_dir"] = os.getcwd()
+            # We need this later to modify the Travis CI configuration
+            # Create the initial (local) repository and commit the
+            #  current state.
+            repo = git.Repo.init(gitdir)
+            allfiles = repo.untracked_files
+            idx = repo.index
+            idx.add(allfiles)
+            # Looks like userdict should include author email....
+            # We need consistent field names here...but we can
+            #  synthesize them in the plugin.
+            # git_name and git_email it is.
+            committer = git.Actor(userdict["github_name"],
+                                  userdict["github_email"])
+            idx.commit("Initial commit.", author=committer,
+                       committer=committer)
+            # Now we need to create the repository at GitHub....
+            #  userdict["github_repo"] must exist.
+            remote_url = create_github_repository(auth, userdict)
+            # Set up remote config to auth correctly
+            chlp = get_git_credential_helper(auth["username"],
+                                             auth["password"])
+            userdict["github_repo_url"] = remote_url
+            origin = repo.create_remote("origin", url=remote_url)
+            cwr = repo.config_writer()
+            if not cwr.has_section("credential"):
+                cwr.add_section("credential")
+            cwr.set("credential", "helper", chlp)
+            cwr.release()
+            # https://gitpython.readthedocs.io/en/stable/tutorial.html
+            #  suggests that you need to wait/sync or something?
+            time.sleep(1)
+            try:
+                os.sync()
```

Comment:
I don't see any wait or sync references in that URL. But if that's what is required to ensure it works, so be it.